invenio_stats/tasks.py

```diff
@@ -38,3 +43,142 @@ def process_events(event_types):
     for e in event_types:
         results.append((e, current_stats.events[e].processor.run()))
     return results
+
+
+@shared_task
+def aggregate_event(event):
+    """Calculate file downloads for today."""
+    file_download_aggregator = StatAggregator(
+        client=current_search_client,
+        event='file_download',
+        time_frame='monthly')
+    file_download_aggregator.run()
+
+
+class StatPreprocessor(object):
+    """Preprocessor class.
+
+    Subclass this class in order to provide specific statistics calculations.
+    """
+
+    def run(self):
+        """Run calculations."""
+        pass
+
+
+class StatAggregator(StatPreprocessor):
+    """Aggregator class."""
+
+    def __init__(self, client, event,
+                 time_frame, query='*'):
+        """Construct aggregator instance."""
+        self.client = client
+        self.event = event
+        self.time_frame = time_frame
+        self.query = query
+        self.aggregation_alias = '{0}-{1}'.format(self.time_frame,
+                                                  self.event)
+
+    def get_bookmark(self):
+        """Get last aggregation date."""
+        if not Index(self.aggregation_alias,
+                     using=self.client).exists():
```

Comment:
I am not sure to understand what you are trying to do here. The alias should always exist.